ci/auto-generate-readme.sh

```diff
@@ -1,45 +1,63 @@
 #!/bin/bash
+
+set -euo pipefail
+
 echo "Auto-generating README files..."
 
-# Search for notebook files named README.ipynb in the ../templates directory
-notebook_files=$(find ../templates -name "README.ipynb")
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+if [[ "$(pwd)" != "${REPO_ROOT}" ]]; then
+	echo "Must run this script at repo's root directory".
+	exit 1
+fi
 
-# Loop through each notebook file
-for notebook_file in $notebook_files; do
-    # Exclude specific notebooks from conversion
-    if [ "$notebook_file" != "../templates/templates/getting-started/README.ipynb" ] && [ "$notebook_file" != "../templates/templates/e2e-llm-workflows/README.ipynb" ] && ! grep -q "Time to complete" $notebook_file; then
-        echo "**********"
-        echo "LINT ERROR: $notebook_file must include 'Time to complete' statement, failing."
-        echo "**********"
-        exit 1
-    fi
-    if [ "$notebook_file" != "../templates/templates/e2e-llm-workflows/README.ipynb" ]; then
-        # Convert notebook file to README.md using nbconvert
-        jupyter nbconvert --to markdown "$notebook_file" --output-dir "$(dirname "$notebook_file")"
-    else
-        echo "Skipping README generation for $notebook_file"
-    fi
-done
+# Search for notebook files named README.ipynb in the ../templates directory
+TEMPLATES_DIRS=($(find "templates" -mindepth 1 -maxdepth 1 -type d))
```

Comment:
just wondering why do we need another pair of parentheses here?

Reply:
the inner one is capturing the output, the outer one converts it into a bash array.

Reply:
oh got it